src/events/balancer.py

```diff
@@ -134,7 +138,8 @@ def _on_config_changed(self, event: EventBase) -> None:
             self.config_manager.set_broker_capacities()
             self.config_manager.set_zk_jaas_config()
 
-            self._on_start(event)
+            self.workload.restart()
+            # self._on_start(event)
```

Comment:
**question:** What motivated this change? Was calling `_on_start` not actually restarting?

Reply:
Yes. For some reason `_on_start` would not restart. I have a feeling that it might still be happening, as only `update_status` is successfully restarting. The correct properties are set long before the restart happens, something is going on that I didn't debug yet. Maybe related with the secret issue.

Reply:
Ah hold on. We are not observing `on_config_changed` event for the balancer. @Batalex should we add `on_config_changed` observer on this method as well?

Reply:
Sorry, I somehow missed the notification for this comment. Yes, we should observe the proper `config-changed` event as well IMO

Reply:
no worries, let me then do that and port these fixes to the k8s PR

Reply:
Do we still need the direct call to `workload.restart`?